src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,3222 @@
+                    { rsTotalAmount = fromProto $ v1 ^. ProtoFields.totalAmount
+                    , rsTotalEncryptedAmount = fromProto $ v1 ^. ProtoFields.totalEncryptedAmount
+                    , rsBakingRewardAccount = fromProto $ v1 ^. ProtoFields.bakingRewardAccount
+                    , rsFinalizationRewardAccount = fromProto $ v1 ^. ProtoFields.finalizationRewardAccount
+                    , rsGasAccount = fromProto $ v1 ^. ProtoFields.gasAccount
+                    , rsFoundationTransactionRewards = fromProto $ v1 ^. ProtoFields.foundationTransactionRewards
+                    , rsNextPaydayTime = snd . fromProto $ v1 ^. ProtoFields.nextPaydayTime
+                    , rsNextPaydayMintRate = fromProto $ v1 ^. ProtoFields.nextPaydayMintRate
+                    , rsTotalStakedCapital = fromProto $ v1 ^. ProtoFields.totalStakedCapital
+                    , rsProtocolVersion = fromProto $ v1 ^. ProtoFields.protocolVersion
+                    }
+            _ -> error "UwU"
+
+data InvokeInstanceInput = InvokeInstanceInput
+    { iiBlockHash :: BlockHashInput
+    , iiInvoker :: Maybe Address
+    , iiInstance :: ContractAddress
+    , iiAmount :: Amount
+    , iiEntrypoint :: Wasm.ReceiveName
+    , iiParameter :: Wasm.Parameter
+    , iiEnergy :: Energy
+    }
+
+instance FromProto Proto.ContractEvent where
+    type Output' Proto.ContractEvent = Wasm.ContractEvent
+    fromProto ce = Wasm.ContractEvent . BSS.toShort $ ce ^. ProtoFields.value
+
+
+instance FromProto Proto.Parameter where
+    type Output' Proto.Parameter = Wasm.Parameter
+    fromProto p = Wasm.Parameter . BSS.toShort $ p ^. ProtoFields.value
+
+instance FromProto Proto.Address where
+    type Output' Proto.Address = Address
+    fromProto addr =
+        case (addr ^? ProtoFields.account, addr ^? ProtoFields.contract) of
+            (Just aAddr, Nothing) -> AddressAccount $ fromProto aAddr
+            (Nothing, Just cAddr) -> AddressContract $ fromProto cAddr
+            _ -> error "nope"
+
+instance FromProto Proto.ContractAddress where
+    type Output' Proto.ContractAddress = ContractAddress
+    fromProto ca = ContractAddress {
+        contractIndex = ContractIndex $ ca ^. ProtoFields.index,
+        contractSubindex = ContractSubindex $ ca ^. ProtoFields.subindex
+    }
+
+instance FromProto Proto.ContractVersion where
+    type Output' Proto.ContractVersion = Wasm.WasmVersion
+    fromProto cv = case cv of
+        Proto.V0 -> Wasm.V0
+        Proto.V1 -> Wasm.V1
+        _ -> error "dangit"
+
+instance FromProto Proto.ContractTraceElement where
+    type Output' Proto.ContractTraceElement = Event
+    fromProto cte =
+        case ( cte ^? ProtoFields.updated
+             , cte ^? ProtoFields.transferred
+             , cte ^? ProtoFields.interrupted
+             , cte ^? ProtoFields.resumed
+             , cte ^? ProtoFields.upgraded
+             ) of
+            (Just updated, Nothing, Nothing, Nothing, Nothing) ->
+                Updated
```

Comment:
As we discussed, this is not the way to go about it. For this one the return would be `Maybe Event` and then

```
fromProto cte = do
    inner <- cte ^. Proto.maybe'element
    case inner of 
        Proto.ContractTraceElement'Element ue -> fromProot ue
        ...
```